cluster-master.js

```diff
@@ -582,41 +582,43 @@ function resize(size, callback) {
   cluster.setMaxListeners(clusterSize * 2);
 
   if (nbWorkers === clusterSize) {
-    resizing = false;
     cb();
     return;
   }
 
-  let thenCnt = 0;
-  function then() {
-    thenCnt++;
-    return then2;
+  function then(worker, thenCb) {
+    startingWorkersCount++;
+    return then2(worker, thenCb);
   }
-  function then2() {
-    if (--thenCnt === 0) {
-      resizing = false;
-      return cb();
-    }
-    return undefined;
+  function then2(worker, then2Cb) {
+    return () => {
+      worker.started = true;
+      startingWorkersCount = startingWorkersCount > 0 ? startingWorkersCount - 1 : 0;
+      if (startingWorkersCount === 0) {
```

Comment:
This condition will only happen if the latest starting worker starts correctly.
If the latest starting worker fails to start, the `startingWorkersCount` will be decremented [here](https://github.com/fasterize/cluster-master/pull/3/commits/abd8eaf75d462e713763e6fc17c5dbfa2cca8ee5#diff-ed99ccfaf553a266d0a8c939c33334c6R350) and thus `then2Cb` will never be called.

I am not sure of the consequences of this.